### PR TITLE
Add `*print-meta*` to allow printing metadata

### PR DESCRIPTION
**Pull request by djblue**

### CODE REVIEW

File: clj/src/cljd/core.cljd [hunk 1]
```diff
@@ -1828,6 +1829,10 @@
   ^:mixin EqualsEquivMixin
   IPrint
   (-print [o ^StringSink sink]
+    (when-let [m (and *print-meta* meta)]
```
Comment (cgrand) [inline clj/src/cljd/core.cljd:1832]: replace by `when-some`, we don't care about false

File: clj/src/cljd/core.cljd [hunk 2]
```diff
@@ -3073,6 +3078,10 @@
                   (.write w sep)
                   (recur xs)))))
           (.write w end))))
+  (when-let [m (and *print-meta* (meta sequence))]
```
Comment (cgrand) [inline clj/src/cljd/core.cljd:3081]: `when-some`

File: clj/src/cljd/core.cljd [hunk 3]
```diff
@@ -4839,6 +4848,10 @@
                   (.write w sep)
                   (recur xs)))))
           (.write w end))))
+  (when-let [m (and *print-meta* (meta m))]
```
Comment (cgrand) [inline clj/src/cljd/core.cljd:4851]: `when-some`

**Review by cgrand — COMMENTED** [review]
I guess `when-let` is used in CLJ/CLJS but there's no point and it simplifies the emitted code to only check for nil.

**Comment by djblue** [comment]
@cgrand when I switch to `when-some` I get test failures around metadata of the form `^false` 🤔

File: clj/src/cljd/core.cljd (hunk 1 above)
Comment (djblue) [reply clj/src/cljd/core.cljd:1832]: Because we also need to check if `*print-meta*` is enabled, we do care about false here.

**Comment by cgrand** [comment]
@djblue I read too hastily
An alternative could be `when-some [m (when *print-meta* meta)]` but nevermind

**Comment by cgrand** [comment]
@djblue merged, thanks

**Pull request merged**